test/Performance/run-perf_tests.py

```diff
@@ -0,0 +1,792 @@
+    Args:
+        conn (obj): A connection to the result database
+        id_field (str): The name of the id column
+        table (str): The name of the table that contains the entry
+        name_field (str): The name of the field to compare the value against
+        value (str): The value that its id is requested
+    Returns:
+        id of the value if the value exists in the table, None otherwise
+    """
+    query = "SELECT {0} FROM {1} WHERE {2}='{3}'"
+    cursor = conn.cursor()
+    cursor.execute( query.format( id_field, table, name_field, value ))
+    id = cursor.fetchone()
+    cursor.close()
+    if id is not None:
+        return id[0]
+    return id
+ 
+def get_id_no_quote( conn, id_field, table, name_field, value ):
+    """
+    This module returns id of an entry when value is not a string. 
+    @TODO This is a hack, could not get binary binding working with pyodbc.
+    This module should be removed and get_id should use binding parameters.
+    Args:
+        conn (obj): A connection to the result database
+        id_field (str): The name of the id column
+        table (str): The name of the table that contains the entry
+        name_field (str): The name of the field to compare the value against
+        value (bin): The value that its id is requested
+    Returns:
+        id of the value if the value exists in the table, None otherwise
+    """
+    query = "SELECT {0} FROM {1} WHERE {2}={3}"
+    cursor = conn.cursor()
+    cursor.execute( query.format( id_field, table, name_field, value ))
+    id = cursor.fetchone()
+    cursor.close()
+    if id is not None:
+        return id[0]
+    return id
+ 
+def get_test_database():
+    """
+    This module reads test database details from connect.php and stores them into an instance of DB class
+    Returns:
+        A DB object that contains database credentials
+    """
+    test_db = DB()
+    for line in open( connect_file ):
+        if "server" in line:
+            test_db.server_name = line.split("=")[1].strip()[1:-2]
+        elif "database" in line:
+            test_db.database_name = line.split("=")[1].strip()[1:-2]        
+        elif "uid" in line:
+            test_db.username = line.split("=")[1].strip()[1:-2]
+        elif "pwd" in line:
+            test_db.password = line.split("=")[1].strip()[1:-2]
+    return test_db
+ 
+def connect( db ):
+    """
+    This module creates a connection to the given database
+    Args:
+        db (obj): database object
+    Returns:
+        A connection object to the given database       
+    """
+    return pyodbc.connect(
+          driver="{ODBC Driver 13 for SQL Server}"
+        , host=db.server_name
+        , database=db.database_name
+        , user=db.username
+        , password=db.password
+        , autocommit = True)
+ 
+def get_server_version( server ):
+    """
+    This module returns the version of the given server
+    Args:
+        server (obj): Server object to connect to

```

Comment:
I am using this class as server when database_name is not required. Example, when fetching @@Version. We could probably name this class better though.